bridge/node/src/api.rs

```diff
@@ -33,33 +43,69 @@ pub fn get_graph_config(mut cx: FunctionContext) -> JsResult<JsObject> {
 /// Create a new graph state
 /// # Arguments
 /// * `cx` - Neon FunctionContext
-/// * `env` - Neon Environment object
+/// * `env` - Neon Environment object extracted from context
 /// # Returns
 /// * `JsResult<JsObject>` - Neon JsObject containing the graph state
 /// # Errors
 /// * Throws a Neon error if the graph state cannot be created
 /// # Safety
-pub fn create_graph_state(mut cx: FunctionContext) -> JsResult<JsObject> {
-	return cx.throw_error("Not implemented")
+pub fn initialize_graph_state(mut cx: FunctionContext) -> JsResult<JsNumber> {
+	let environment_obj = cx.argument::<JsObject>(0)?;
+	let rust_environment = unsafe { environment_from_js(&mut cx, environment_obj) };
+	let graph_state = GraphState::new(rust_environment);
+
+	// Generate a unique identifier for the graph state
+	let graph_state_id = {
+		let mut next_id = NEXT_GRAPH_STATE_ID.lock().unwrap();
+		let id = *next_id;
+		*next_id += 1;
```

Comment:
Maybe use `.wrapping_add(1)` here? Reconnection service will ideally be long-running, and as we load/unload user  graphs we'll burn through ids at a rate of potentially `(num_users)**2`. (Realizing it's unlikely we'll overflow a `usize` before that app is restarted at some point, but it's probably even less likely that we'll have a key conflict if we wrap (since our intended usage pattern for graph is to load a graph, do stuff with it, then remove it from the cache at some point)

Reply:
Thats a great suggestion